### ISSUE TITLE
Loading custom trained YOLO-NAS using ultralytics give NoneType' object has no attribute 'fuse error

### ISSUE THREAD

**Issue by james-imi** [ultralytics/ultralytics#8911]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have loaded a custom trained YOLO-NAS from super-gradients.
When loading using Ultralytics, I get this error

```
Cell In[38], [line 4](vscode-notebook-cell:?execution_count=38&line=4)
      [1](vscode-notebook-cell:?execution_count=38&line=1) from ultralytics import NAS
      [3](vscode-notebook-cell:?execution_count=38&line=3) # Load a COCO-pretrained YOLO-NAS-s model
----> [4](vscode-notebook-cell:?execution_count=38&line=4) model = NAS('experiments/yolo_nas_m-df-adamtest2/RUN_20240313_131741_472534/ckpt_best.pth')
      [6](vscode-notebook-cell:?execution_count=38&line=6) # Display model information (optional)
      [7](vscode-notebook-cell:?execution_count=38&line=7) model.info()

File [~/anaconda3/envs/nas/lib/python3.9/site-packages/ultralytics/models/nas/model.py:49](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/ultralytics/models/nas/model.py:49), in NAS.__init__(self, model)
     [47](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/ultralytics/models/nas/model.py:47) """Initializes the NAS model with the provided or default 'yolo_nas_s.pt' model."""
     [48](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/ultralytics/models/nas/model.py:48) assert Path(model).suffix not in (".yaml", ".yml"), "YOLO-NAS models only support pre-trained models."
---> [49](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/ultralytics/models/nas/model.py:49) super().__init__(model, task="detect")

File [~/anaconda3/envs/nas/lib/python3.9/site-packages/ultralytics/engine/model.py:142](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/ultralytics/engine/model.py:142), in Model.__init__(self, model, task, verbose)
    [140](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/ultralytics/engine/model.py:140)     self._new(model, task=task, verbose=verbose)
    [141](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/ultralytics/engine/model.py:141) else:
--> [142](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/ultralytics/engine/model.py:142)     self._load(model, task=task)
    [144](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/ultralytics/engine/model.py:144) self.model_name = model

File [~/anaconda3/envs/nas/lib/python3.9/site-packages/torch/autograd/grad_mode.py:27](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/torch/autograd/grad_mode.py:27), in _DecoratorContextManager.__call__.<locals>.decorate_context(*args, **kwargs)
     [24](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/torch/autograd/grad_mode.py:24) @functools.wraps(func)
     [25](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/torch/autograd/grad_mode.py:25) def decorate_context(*args, **kwargs):
     [26](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/torch/autograd/grad_mode.py:26)     with self.clone():
---> [27](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/torch/autograd/grad_mode.py:27)         return func(*args, **kwargs)

File [~/anaconda3/envs/nas/lib/python3.9/site-packages/ultralytics/models/nas/model.py:62](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/ultralytics/models/nas/model.py:62), in NAS._load(self, weights, task)
     [60](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/ultralytics/models/nas/model.py:60)     self.model = super_gradients.training.models.get(weights, pretrained_weights="coco")
     [61](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/ultralytics/models/nas/model.py:61) # Standardize model
---> [62](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/ultralytics/models/nas/model.py:62) self.model.fuse = lambda verbose=True: self.model
     [63](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/ultralytics/models/nas/model.py:63) self.model.stride = torch.tensor([32])
     [64](https://file+.vscode-resource.vscode-cdn.net/home/james/workspace/Adra/super-gradients/~/anaconda3/envs/nas/lib/python3.9/site-packages/ultralytics/models/nas/model.py:64) self.model.names = dict(enumerate(self.model._class_names))
```

with the code
```
from ultralytics import NAS

# Load a COCO-pretrained YOLO-NAS-s model
model = NAS('experiments/yolo_nas_m-df-adamtest2/RUN_20240313_131741_472534/ckpt_best.pth')

# Display model information (optional)
model.info()
```

I have 7 classes in the customt trained model.
